gh-pr-context.py

```
@@ -88,16 +88,79 @@ def gh_api_jq(endpoint, jq_filter):
+    work_tree = os.getcwd()
+    while work_tree != os.path.dirname(work_tree):
+        git_file = os.path.join(work_tree, ".git")
+        if os.path.isfile(git_file):
+            break
+        work_tree = os.path.dirname(work_tree)
+    else:
+        return  # no .git file found
+
+    git_file = os.path.join(work_tree, ".git")
+    try:
+        content = open(git_file).read().strip()
```

Ruff (0.15.15)

[warning] Use a context manager for opening files

(SIM115)